tgui/packages/tgui/interfaces/ChemHeater.js

```diff
@@ -0,0 +1,123 @@
+import { useBackend } from "../backend";
+import { AnimatedNumber, Box, Button, Dimmer, Icon, Knob, Section, SectionWithTitleOverflow, Stack } from '../components';
+import { Window } from '../layouts';
+import { freezeTemperature, getTemperatureColor, getTemperatureIcon, getTemperatureChangeName } from './common/temperatureUtils.js';
+import { NoContainer, ReagentGraph, ReagentList } from './common/ReagentInfo.js';
+
+export const ChemHeater = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { containerData, isActive, targetTemperature } = data;
+
+  return (
+    <Window
+      title="Reagent Heater/Cooler"
+      width={320}
+      height={385}>
+      <Window.Content>
+        <ChemDisplay container={containerData} targetTemperature={targetTemperature} active={isActive} />
+        <Section title="Temperature Control">
+          <Stack align="center">
+            <Stack.Item>
+              <Knob
+                animated
+                size={2}
+                value={targetTemperature}
+                minValue={0}
+                maxValue={1000}
+                format={value => value + " K"}
+                onDrag={(e, value) => act('adjustTemp', { temperature: value })}
+              />
+            </Stack.Item>
+            <Stack.Item grow basis={0} overflow="hidden">
+              <Box
+                className="ChemHeater__TemperatureNumber"
+                nowrap
+                p={1}
+                fontSize={1.5}
+                color={getTemperatureColor(targetTemperature)}
+                backgroundColor="black">
+                <Box fontSize={1}>Target</Box>
+                <Icon name={getTemperatureIcon(targetTemperature)} pr={0.5} />
+                <AnimatedNumber value={targetTemperature} /> K
+              </Box>
+            </Stack.Item>
+            <Stack.Item basis={9.6} align="center">
+              <Button
+                icon="power-off"
+                disabled={!containerData?.totalVolume}
+                color={isActive ? "red" : "green"}
+                fluid
+                height="100%"
+                fontSize={1.25}
+                textAlign="center"
+                onClick={() => act(isActive ? 'stop' : 'start')}>
+                {isActive ? "Deactivate" : "Activate"}
+              </Button>
+            </Stack.Item>
+          </Stack>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ChemDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const { active = false, targetTemperature = freezeTemperature } = props;
+  const container = props.container ?? NoContainer;
+  const working = active && !container.fake;
+  const { temperature, totalVolume } = container;
+
+  return (
+    <SectionWithTitleOverflow
+      title={
+        <Box as="span"
+          style={{
+            "text-transform": "capitalize",
```

Comment:
Not requiring for this PR:

Assuming intent here is to convert lowercase obj names to have uppercase first letters. This seems like useful-enough behavior to wrap into a prop on the component, or at least with a util function, rather than do via a CSS style.

Reply:
Added as a property to SectionWithTitleOverflow, which I have renamed SectionEx.